tests/integration/pilot/routing_test.go

```diff
@@ -230,6 +230,7 @@ func vmTestCases(vm echo.Instance) []TrafficTestCase {
 			name: "dns: VM to k8s headless service",
 			from: vm,
 			to:   apps.headless,
+			host: apps.headless.Config().FQDN() + ".",
```

Comment:
can you please revert these changes? the dot thing is nice but its masking the underlying problem.. or wait for mine to pass?